cfme/tests/infrastructure/test_infra_quota.py

```diff
@@ -114,7 +113,7 @@ def template_name(provisioning):
 @pytest.fixture(scope="function")
 def provisioner(request, setup_provider, provider):
     def _provisioner(template, provisioning_data, delayed=None):
-        vm = Vm(name=vm_name, provider=provider, template_name=template)
+        vm = provider.vm_type(name=vm_name, provider=provider, template_name=template)
```

Comment:
Looks good, though I have an idea - Can you create a method in the provider that will use the vm_type and also injectthe provider itself?

So you could then just do:
```python
provider.instantiate_vm(name=vm_name, template_name=template)
```